stan/math/rev/core/init_chainablestack_tbb.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef STAN_MATH_REV_CORE_INIT_CHAINABLESTACK_TBB_HPP
+#define STAN_MATH_REV_CORE_INIT_CHAINABLESTACK_TBB_HPP
+
+// TODO(SW): remove STAN_THREADS guard once Intel TBB is fully
+// mandatory
+#ifdef STAN_THREADS
+
+#include <stan/math/rev/core/chainablestack.hpp>
+
+#include <tbb/task_scheduler_observer.h>
+
+#include <mutex>
+#include <iostream>
+#include <unordered_map>
+#include <thread>
+#include <utility>
+#include <tuple>
+
+namespace stan {
+namespace math {
+
+static std::mutex cout_mutex;
+
+// initialize AD tape for TBB threads
+struct ad_tape_observer : public tbb::task_scheduler_observer {
```

Comment:
General docs would be nice. With this stuff especially I, Rok, pretty much any other reviewer will be reading a lot of these methods and schemes for the first time. So a little doc here is nice to lay down intent

Reply:
Sure... I was in a hurry yesterday night... can add some URL refs to TBB doc, for example.